File: 566. Reshape the Matrix/566. Reshape the Matrix.go
func matrixReshape(mat [][]int, r int, c int) [][]int {
    if mat == nil{
        return nil
    }
    n := len(mat)
    m := len(mat[0])
    if n * m != r * c {
        return mat
    }
    ans := make([][]int, r)
    for i := 0; i < n; i++{
        for j := 0; j < m; j++{
            tmp := i * m + j
            var x int = tmp / c
            var y int = tmp % c
            if y == 0{
                ans[x] = make([]int, c)
            }
            ans[x][y] = mat[i][j]
        }
    }
    return ans
}

File: 566. Reshape the Matrix/566. Reshape the Matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if not mat:
            return
        n = len(mat)
        m = len(mat[0])
        if n*m != r*c:
            return mat
        ans = [[0 for _ in range(c)] for _ in range(r)]
        for i in range(n):
            for j in range(m):
                tmp = i*m + j
                x = int(tmp/c)
                y = int(tmp%c)
                ans[x][y] = mat[i][j]
        return ans
                
